src/consts.rs

```diff
@@ -23,6 +26,63 @@ enum_from_primitive! {
     }
 }
 
+impl fmt::Display for ZkError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            ZkError::APIError => write!(f, "Zookeeper Error: {}", "APIError"),
```

Comment:
I would replace the whole match statement with the following line `write!(f, "Zookeeper Error: {}", self.description())` in order to reduce boilerplate. You will have to change `use std::error;` to `use std::error::Error;` to make it work, because of trait visibility.